commit 92fefacff4842c261d2741c55c0bfa4d18ffcd7f (refs/changes/71/4971/1)
Author: Trent Mick <trentm@gmail.com>
Date:   2018-10-17T12:41:37-07:00 (12 months ago)
    
    TRITON-886 Add support for `channel` option to IMGAPI client `getImage` method.

diff --git a/CHANGES.md b/CHANGES.md
index ba834df..92840a0 100644
--- a/CHANGES.md
+++ b/CHANGES.md
@@ -10,7 +10,8 @@
 
 ## 12.0.0
 
-- TRITON-742 Remove dsapi client
+- TRITON-886 Add support for `channel` option to IMGAPI `getImage` method.
+- [Backward incompatible] TRITON-742 Remove dsapi client
 
 ## 11.3.1
 
diff --git a/lib/imgapi.js b/lib/imgapi.js
index db15d13..feb1bb6 100644
--- a/lib/imgapi.js
+++ b/lib/imgapi.js
@@ -693,6 +693,9 @@ IMGAPI.prototype.listImages = function listImages(filters, options, callback) {
  *      - headers {Object} Optional extra request headers.
  *      - inclAdminFields {Boolean} Whether to include admin fields in the
  *        returned image.
+ *      - channel {String} Optional override for the channel set on the
+ *        constructor. This is only relevant for IMGAPI servers that
+ *        support channels.
  * @param {Function} callback : `function (err, image, res)`
  */
 IMGAPI.prototype.getImage =
@@ -715,6 +718,7 @@ function getImage(uuid, account, options, callback) {
     assert.object(options, 'options');
     assert.optionalObject(options.headers, 'options.headers');
     assert.optionalBool(options.inclAdminFields, 'options.inclAdminFields');
+    assert.optionalString(options.channel, 'options.channel');
     assert.func(callback, 'callback');
     var inclAdminFields = (options.inclAdminFields !== undefined
             ? options.inclAdminFields.toString()
@@ -722,7 +726,7 @@ function getImage(uuid, account, options, callback) {
 
     var path = self._path('/images/' + uuid, {
         account: account,
-        channel: self.channel,
+        channel: options.channel || self.channel,
         inclAdminFields: inclAdminFields
     });
     self._getAuthHeaders(function (hErr, headers) {
